content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md

```diff
@@ -12,16 +12,28 @@ CatalogContent:
   - 'paths/machine-learning-engineer'
 ---
 
-**Convolutional Neural Networks** (CNNs) are a type of neural network that are primarily used for image classification. They are also used for other tasks such as object detection, image segmentation, and natural language processing.
+**Convolutional Neural Networks** (CNNs) are a type of neural network that are primarily used for computer vision tasks, such as image classification, object detection, and semantic segmentation. CNNs are highly effective for [machine learning](https://www.codecademy.com/resources/docs/ai/machine-learning) tasks that involve structured grid-like data, such as images. However, they can be used in other areas of machine learning, such as natural language processing or audio signal processing.
 
-CNNs are inspired by the visual cortex of the brain. The visual cortex is the part of the brain that processes visual information.
+## What is Convolution?
+
+Convolution is a mathematical operation which involves combining two functions to produce a third function. It essentially “blends” or “merges” two functions together. For instance, if you have two functions, _f(x)_ and _g(x)_, convolution can be employed to combine them and generate a new function, denoted as _(f \* g)(x)_. This new function captures the combined impact of the original functions, illustrating their interaction and overlap.
 
 ## How Does a CNN Work?
 
-CNNs extract features from images using specialized layers called convolutional layers. These layers apply filters to the input image which helps them to learn about features such as patterns and edges.
+Convolution in CNNs allows the network to automatically learn and extract visual features from images, model spatial relationships, handle variations in object position and appearance, and capture meaningful relationships within images. Important characteristics in images can be captured using CNNs, such as edges, corners, textures, and shapes.
+
+In a CNN's convolutional filter context, _f(x)_ represents the input data, and _g(x)_ represents the convolutional filter used in the network. The input data is a small matrix typically from an image or a feature map produced by a previous layer in the network, and the convolutional filter is a small matrix of weights that was learned in the training process. The convolutional filter essentially acts as a window that scans the input data to extract relevant features.
+
+During convolution, the filter (_g(x)_) is slid over the input data (_f(x)_). At each position, the filter performs element-wise multiplication of the two matrices. The products are then summed to produce a single value which is the output of the convolution in that position. This process is repeated for all positions, which results in an “output feature map,” which is a two-dimensional representation of the activations produced by applying the convolutional filters to the input data.
```

Comment:
Is it possible to show an example of this using a 2X2 matrix?

Reply:
Do you have a recommendation of how this example might look? 😅 The description I provided is VERY simplified but illustrates a multi-step process with complicated steps, and I'm having a hard time coming up with an image that can summarize this while maintaining the simplicity of the description. (Or at least one that I know how to make)

My first thought was to have the example show an image, then pull out a rectangle of it and have that be the matrix (f(x)), and then have another matrix (g(x)) to show the matrix multiplying, but I'm not sure how to actually make something like that? (I usually use draw.io, but I'm open to trying another free tool if you have a recommendation!)

 I searched for some other images on how CNNs work, but most of them depict either a large overview of how CNNs work (with additional info I haven't gone into in this article), or are separated into multiple images with more complicated descriptions.

Reply:
It doesn't have to be an image. The convolution operation can be shown using a 2X2 matrix over a 5X5 matrix for example. Something like this: https://medium.com/@ramitag18/performing-convolution-on-a-matrix-4682fd364591

Reply:
You could include pictures of convolving two matrices together. 😁 LMK if this works. If it is too complicated, we can do without it.

Reply:
@HishamT Ok! I think I'd rather do without it, if that still works for you? Not because I don't like the idea, just because it's hard to make 😅

Reply:
That's fine. 😃